.github/workflows/pull-build.yml

```diff
@@ -0,0 +1,47 @@
+name: Build and test library
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: windows-latest
+
+    steps:
+    - run: mkdir -p artifactApp
+    - run: mkdir -p artifactTests
+
+    - uses: actions/checkout@master
+
+    - uses: actions/checkout@v2
+    - name: Setup .NET Core
+      uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: 3.1.101
+   
+    - name: Install dependencies
+      run: dotnet restore Selenium.Axe/Selenium.Axe.sln
+   
+    - name: Build
+      run: dotnet build Selenium.Axe/Selenium.Axe.sln --configuration Release --no-restore
+    
+    - name: Update Chrome
```

Comment:
In general, `geckodriver` and `chromedriver` only support running against the specific versions of Firefox/Chrome they were build/tested against, so this makes me a bit nervous; it will be installing a version of chrome/firefox that isn't synced up with any installed ChromeDriver/GeckoDriver.

I think the two options that would make the most sense to consider would be:
1. Don't install any chrome/firefox version and use the version pre-installed on the windows-latest agent (the agents come pre-installed with synchronized versions of chrome+chromedriver and firefox+geckodriver). I believe that's what the existing tests are already doing in the old Pipelines builds (the environment variables for the webdriver paths that the test setup method looks for are set by the agent setup)
2. Install *both* the browsers and the webdrivers, and pin the specific installed versions to ensure they stay in sync; that would amount to pinning a version number in these commands that matches the version numbers of the chromedriver/geckodriver nuget packages used in the test project .csproj, and then removing the test setup logic that prefers the environment variables if they exist.

I would lean towards option 1 being good-enough and a little easier to manage (because you don't have to update version numbers in this file manually without dependabot's help), but I think both of these approaches would be reasonable

Reply:
I see this a bit differently.
In my experience most users are using the latest version of Chrome, Firefox, Edge, whatever.
If our user base is largely going to be on the latest version than I think it benefits the project to stay on the latest version. Over on the [MAQS ](https://github.com/Magenic/MAQS/) project this has really helped my team find issue quickly, often before our user base.
Admittedly in the rare instances where the latest browser's driver isn't available yet we would need to have Choco update to a specific version, but it seems like the benefits out way the cost.

Reply:
I definitely agree that keeping dependencies up to date is important; if you'd like to keep more up to date than the build agent versions allow for, but still make sure that the browser and webdriver versions stay in sync, I think the main options would be:

* Pin the versions of both, but make it a policy to manually update the version of the browser whenever dependabot opens a PR with a new version of a webdriver package; this should keep the browser as up to date as possible while still keeping the webdrivers in sync
* Put the responsibility of keeping them in sync on an external service (the SauceLabs option you suggested in a separate email)

The second option is probably lower maintenance; there is some risk of SauceLabs deciding to change their "free for OSS" policy someday, but the testing strategy can always be changed later if that ever becomes a problem. I think experimenting with that would be a good approach for keeping continuously up to date.

Reply:
I don't think it's necessary to hold up the rest of the PR on this discussion either way, so I'm approving my review with this question still open; it can be iterated on in a separate PR